LearningMissionSimulation/ObjectGenerator.cs

```diff
@@ -115,10 +115,11 @@ public static List<string> GenerateRecommendationList()
         }
 
         // Creates a random list of Classroom objects
-        public static Classroom GenerateClassroom()
+        public static Classroom GenerateClassroom(Module module)
         {
-            //not implemented yet!
-            return null;
+            return new Classroom(maximumCapacity: SimulationConstants.MaxClassroomSize, minimumCapacity: SimulationConstants.MinClassroomSize,
+                                 module: GenerateModule(GenerateSubject().Id), schedule: GenerateSchedule(), name: AttributeGenerator.GetUnitName(UnitType.Classroom),
```

Comment:
Your function takes in the module parameter. The purpose of the module parameter is to use it in the Classroom object generation. You should NOT generate a new module, you should use the Module object that we pass in the method as an input parameter:
Line 121: module: GenerateModule(GenerateSubject().Id) - INCORRECT!
should be : module: module